File: php/get_categories.php
#!/usr/bin/env php
<?php

require_once 'api.php';

$api_key = $argv[1];
$access_token = $argv[2];

if (!$api_key || !$access_token) {
    echo "Usage: get_categories.php <api_key> <access_token>\n";
    exit();
}

$api = new VaultREAPI($api_key, $access_token);

// Get contact categories
list($code, $result) = $api->get('/categories/contact?sort=name');

if ($code == 403) {
    echo "HTTP 403: Invalid API Key";
    exit();
} elseif ($code == 401) {
    echo "HTTP 401: Invalid bearer token\n";
    exit();
}

$result = json_decode($result);

foreach (@$result->{"items"} as $item) {

    echo $item->{"id"} . ": " . $item->{"name"} . "\n";

}

?>

File: perl/get_categories.pl
#!/usr/bin/perl

use strict;
use warnings;

use FindBin;
use File::Spec;
use lib File::Spec->catdir($FindBin::Bin, '.');

use VaultREAPI;

use JSON;

my $api_key = $ARGV[0];
my $bearer_token = $ARGV[1];

if (!$api_key || !$bearer_token) {
    die "Usage: $0 <api_key> <bearer_token>\n";
}

my $api = new VaultREAPI($api_key, $bearer_token);

# Get user
my ($code, $result) = $api->get('/categories/contact?sort=name');

print "HTTP Response Code: " . $code . "\n";

if ($result) {

    $result = decode_json($result);

    foreach my $item (@{$result->{"items"}}) {
        print $item->{"id"} . ": " . $item->{"name"} . "\n";
    }

}


exit;
